src/CSVMsg.chpl

```diff
@@ -353,6 +353,110 @@ module CSVMsg {
+                lineNumber=getLineNumber(),
+                routineName=getRoutineName(),
+                moduleName=getModuleName(),
+                errorClass="InvalidArgumentError");
+        const intersection = domain_intersection(A.localSubdomain(), filedom);
+        if(intersection.size == 0) {
+            // There is nothing to be done for this file on this locale
+            return;
+        }
+
+        var fr = openReader(filename);
+        if(hasHeaders) {
+            var line:string;
+            // The first three lines are headers we don't care about
+            // Advance through them without reading them in
+            for param i in 0..<3 {
+                try {fr.advanceThrough(b'\n');}
+                catch {
+                    throw getErrorWithContext(
+                        msg="This CSV file is missing header values when headers were detected. This error should not arise.",
+                        lineNumber=getLineNumber(),
+                        routineName=getRoutineName(),
+                        moduleName=getModuleName(),
+                        errorClass="IOError");
+                }
+            }
+        }
+        var colIdx = -1;
+        // This next line will have the columns
+        for (column,i) in zip(fr.readLine(stripNewline = true).split(colDelim), 0..) {
+            if column == colName {
+                colIdx=i;
+                break;
+            }
+        }
+
+        if(colIdx == -1) then
+            throw getErrorWithContext(
+                msg="The dataset %s was not found in %s".doFormat(colName, filename),
+                lineNumber=getLineNumber(),
+                routineName=getRoutineName(),
+                moduleName=getModuleName(),
+                errorClass="DatasetNotFoundError");
+
+        var line = new csvLine(t, colIdx, colDelim);
+        for x in intersection {
+            try {
+                fr.read(line);
+            } catch e: BadFormatError {
+                // Handle errors
+                if t!=string {
+                    if allowErrors {
+                        A[x] = min(t);
+                        continue;
+                    } else {
+                        hadError |= true;
+                    }
+                } else {
+                    A[x]="";
+                }
+            }
+            // store the items from the csvLine in the array
+            A[x] = line.item;
+        }
+    }
+
+    use Regex;
+    record csvLine /*: readDeserializable*/{ // Re-add this when dropping support for 1.31
+
+        type itemType;
+        var item: itemType;
+        const colIdx: int;
+        const colDelim: string;
+        const r: regex(string);
+
+        proc init(type itemType, colIdx: int, colDelim: string) {
+            this.itemType = itemType;
+            this.colIdx = colIdx;
+            this.colDelim = colDelim;
+            this.r = try! new regex("[\n"+this.colDelim+"]");
```

Comment:
We only need this regex when it's a string type, right? I'm not sure it's a big deal at all, just wondering if we could potentially avoid creating this everytime.

Also, I am seeing the `try!` syntax here, which we try to avoid in Arkouda. Even if it is very unlikely to throw, we never want to create a scenario where we can crash the server. We could possibly move it to after a `init this` (I think that's the syntax...) or something like that to be able to throw?

Reply:
Yes, we would only need this when the type being read in is a `string`. I can make it so that it doesn't compile the regex when we aren't read a `string`.

I will make the regex compilation throwing as well.